packages/cli/src/config/config.integration.test.ts

```diff
@@ -240,4 +240,320 @@ describe('Configuration Integration Tests', () => {
       expect(config.getExtensionContextFilePaths()).toEqual(contextFiles);
     });
   });
+
+  describe('allowCommands Configuration', () => {
+    it('should handle undefined allowCommands', () => {
+      const configParams: ConfigParameters = {
+        cwd: '/tmp',
+        contentGeneratorConfig: TEST_CONTENT_GENERATOR_CONFIG,
+        embeddingModel: 'test-embedding-model',
+        sandbox: false,
+        targetDir: tempDir,
+        debugMode: false,
+        sessionId: 'test-session',
+        model: 'test-model',
+      };
+
+      const config = new Config(configParams);
+      expect(config.getAllowCommands()).toBeUndefined();
+    });
+
+    it('should load allowCommands from configuration', () => {
+      const configParams: ConfigParameters = {
+        cwd: '/tmp',
+        contentGeneratorConfig: TEST_CONTENT_GENERATOR_CONFIG,
+        embeddingModel: 'test-embedding-model',
+        sandbox: false,
+        targetDir: tempDir,
+        debugMode: false,
+        sessionId: 'test-session',
+        model: 'test-model',
+        allowCommands: ['ls', 'pwd', 'git*'],
+      };
+
+      const config = new Config(configParams);
+      expect(config.getAllowCommands()).toEqual(['ls', 'pwd', 'git*']);
+    });
+
+    it('should handle empty allowCommands array', () => {
+      const configParams: ConfigParameters = {
+        cwd: '/tmp',
+        contentGeneratorConfig: TEST_CONTENT_GENERATOR_CONFIG,
+        embeddingModel: 'test-embedding-model',
+        sandbox: false,
+        targetDir: tempDir,
+        debugMode: false,
+        sessionId: 'test-session',
+        model: 'test-model',
+        allowCommands: [],
+      };
+
+      const config = new Config(configParams);
+      expect(config.getAllowCommands()).toEqual([]);
+    });
+
+    it('should handle various pattern types in allowCommands', () => {
+      const configParams: ConfigParameters = {
+        cwd: '/tmp',
+        contentGeneratorConfig: TEST_CONTENT_GENERATOR_CONFIG,
+        embeddingModel: 'test-embedding-model',
+        sandbox: false,
+        targetDir: tempDir,
+        debugMode: false,
+        sessionId: 'test-session',
+        model: 'test-model',
+        allowCommands: [
+          'ls', // exact match
+          'git*', // glob pattern
+          '*.sh', // glob pattern
+          'test?', // glob pattern
```

Comment:
Missing tests for `foo[0-9]` and `foo{bar,baz}` style of globbing.

Reply:
+1